windows: Automatically find Windows SDK when building gpui

ch45er opened the pull request:
### Summary

This PR changes `gpui/build.rs` to look up the Windows SDK directory in the registry instead of falling back to a hard-coded path.

---

### Problem

Currently, building `gpui` on Windows requires `fxc.exe` to be in `PATH` or at a predefined location (unless `GPUI_FXC_PATH` is set). This requires to maintain a certain build environment with proper paths/vars or to install the specific SDK version.

It is possible to find the SDK automatically using the registry keys it creates upon installation. Specifically in
`SOFTWARE\\WOW6432Node\\Microsoft\\Microsoft SDKs\\Windows\\v10.0` branch there are:
* `InstallationFolder` telling the SDK installation location;
* `ProductVersion` telling the SDK version in use.

These keys provide enough information to locate the SDK binaries, with added robustness:
* handles non-standard SDK installation path;
* deterministically selects the latest SDK when multiple versions are present.

---

### Changes Made

* **Updated `crates/gpui/build.rs`**:

  * added dependency on `winreg`
  * introduced `find_latest_windows_sdk_binary()` helper
  * updated fallback logic to use registry lookup

This PR only changes the fallback location, and does not touch the established environment-based workflow.

Release Notes:

- N/A

---

### Impact

Reduces manual configuration needed to build GPUI on Windows.

ch45er committed:
windows: Robust location of Windows SDK.

Use registry lookup to locate path to the latest Windows SDK binaries
instead of hard-coded fallback.

localcc commented on crates/gpui/build.rs:
```diff
@@ -324,6 +325,21 @@ mod windows {
+            let product_version: String = key.get_value("ProductVersion").unwrap_or_default(); // "10.0.22621" NOTE: does not include trailing `.0`
+
+            if !install_folder.is_empty() && !product_version.is_empty() {
+                return format!("{}bin\\{}.0\\x64\\", install_folder, product_version);
```
there's a very rare possibility of the version having the patch number in which case the .0 would theoretically break, e.g. 10.0.22621.1778, ideally a check would be added to ensure we don't create a non existing path

ch45er committed:
windows: Support patch versions of the SDK.

Add heuristic to support non-.0 patch revisions.
Add file existence checks to avoid returning invalid paths.
Replace empty-string sentinel with idiomatic Option<PathBuf>.
Use Path/PathBuf operations instead of string concatenation.
Propagate I/O and registry errors via Result.

reflectronic committed:
Simplify

reflectronic committed:
Oops

reflectronic committed:
Oops
